packages/common/infra/src/page/record.ts

```diff
@@ -0,0 +1,65 @@
+import type { PageMeta } from '@blocksuite/store';
+import { Observable } from 'rxjs';
+
+import { LiveData } from '../livedata';
+import type { Workspace, WorkspaceLocalState } from '../workspace';
+
+export type PageMode = 'edgeless' | 'page';
+
+export class PageRecord {
+  constructor(
+    public readonly id: string,
+    private readonly workspace: Workspace,
+    private readonly localState: WorkspaceLocalState
+  ) {}
+
+  meta = LiveData.from<PageMeta>(
+    new Observable(subscriber => {
+      const emit = () => {
+        const meta = this.workspace.blockSuiteWorkspace.meta.pageMetas.find(
+          page => page.id === this.id
+        );
+        if (meta === undefined) {
+          return;
+        }
+        subscriber.next(meta);
+      };
+
+      emit();
+
+      const dispose =
+        this.workspace.blockSuiteWorkspace.meta.pageMetasUpdated.on(
+          emit
+        ).dispose;
+      return () => {
+        dispose();
+      };
+    }),
+    {
+      id: this.id,
+      title: '',
+      tags: [],
+      createDate: 0,
+    }
```

Comment:
It seems to me the default value can be omitted since there is a `emit` call in the observable body

Reply:
Although there is an emit call, if pageMeta does not exist, this default value will still be used.